processing/src/main/java/org/apache/carbondata/processing/loading/sort/unsafe/merger/UnsafeIntermediateMerger.java

```diff
@@ -111,18 +108,15 @@ public void addFileToMerge(File sortTempFile) {
   }
 
   public void startFileMergingIfPossible() {
-    File[] fileList = null;
-    synchronized (lockObject) {
-      if (procFiles.size() >= parameters.getNumberOfIntermediateFileToBeMerged()) {
+    File[] fileList;
+    if (procFiles.size() >= parameters.getNumberOfIntermediateFileToBeMerged()) {
+      synchronized (lockObject) {
         fileList = procFiles.toArray(new File[procFiles.size()]);
         this.procFiles = new ArrayList<File>();
-        if (LOGGER.isDebugEnabled()) {
-          LOGGER
-              .debug("Submitting request for intermediate merging no of files: " + fileList.length);
-        }
       }
-    }
-    if (null != fileList) {
+      if (LOGGER.isDebugEnabled()) {
+        LOGGER.debug("Sumitting request for intermediate merging no of files: " + fileList.length);
+      }
       startIntermediateMerging(fileList);
```

Comment:
no need to check not null as the old code?

Reply:
No Its nor required as in first if check we are checking the size of the list is greater than number of intermediate file size

Reply:
ok